packages/replay-next/src/suspense/SourceOutlineCache.ts

```diff
@@ -0,0 +1,14 @@
+import { SourceId, getSourceOutlineResult } from "@replayio/protocol";
+import { createCache } from "suspense";
+
+import { ReplayClientInterface } from "shared/client/types";
+
+export const sourceOutlineCache = createCache<
+  [replayClient: ReplayClientInterface, sourceId: SourceId | undefined],
+  getSourceOutlineResult
+>({
+  debugLabel: "sourceOutlineCache",
+  getKey: ([replayClient, sourceId]) => sourceId ?? "",
+  load: ([replayClient, sourceId]) =>
+    sourceId ? replayClient.getSourceOutline(sourceId) : { functions: [], classes: [] },
+});
```

Comment:
Unlike hooks, Suspense caches can be called conditionally during render. This cache should _require_ a source ID and it should be on the caller to decide what to do if one is not provided.
```suggestion
export const sourceOutlineCache = createCache<
  [replayClient: ReplayClientInterface, sourceId: SourceId],
  getSourceOutlineResult
>({
  debugLabel: "sourceOutlineCache",
  getKey: ([replayClient, sourceId]) => sourceId,
  load: ([replayClient, sourceId]) => replayClient.getSourceOutline(sourceId),
});
```

Reply:
The problem was that this cache is used with the `useImperativeCacheValue` hook in the `QuickOpenModal` - and that shouldn't block on fetching the source outline.